### DISCUSSION TITLE
Bogus suggestions endlessly

### DISCUSSION THREAD

**Discussion by jhancock4d** [community/community#7917] [Copilot Conversations]
(I'm using VS Code but I'm pretty sure this would be true anywhere)
I should never get a suggestion that doesn't actually compile. That is, I'm getting constant suggestions that are invalid code, reference properties that don't exist on the objects that the suggestion is referencing and worse.  

And it's constantly making suggestions in VS code and auto-completing even though the brackets etc that it's suggesting would make the code not compile and suggesting adding , in brackets that would cause the code to be invalid in C# as an example (would be fine in Dart which I also use).

All of this conspires to make this plugin maddening to use.  The VS.NET equivalent of this doesn't do any of these things and thus makes it vastly more useful there.

**Comment by MulleDK19** [comment]
Wrong. If it didn't complete with non-existent stuff it wouldn't be very useful at all.
You need to condition it right.

**Comment by jhancock4d** [comment]
@MulleDK19 Um... I would think that if you have a .cs file open in VS Code that it would only make suggestions that would compile against C#. It does not.  I don't know why you'd ever have to condition it to know what language since the file extension tells it at least that.

In fact, one would assume to be useful that it would know about .sln and .csproj files as well and would be reading the entire code base for the project, all nuget packages included and know exactly what properties were available on what objects either using the project system or using the intellisense that is generated from omnisharp.  It does not.

VS.net's version does.  Vs code's doesn't.  And if it does but there are hidden settings, then it's not documented well, and honestly it's rediculous that you should have to, because all of the information is there for, at the very least, it to only make suggestions that can actually compile on the project you're actually working on.

Maybe it's because there are 2 versions of VS Code running, one with a flutter project and one with a C# project (pretty standard C# for backend, flutter for front stuff here...)?

In any event, the current state is unusable even for testing.
